client/src/Components/Mentors.js

```diff
@@ -1,143 +1,138 @@
 import React, { useEffect, useState } from "react";
-import {
-	BrowserRouter as Router,
-	Switch,
-	Route,
-	Link,
-} from "react-router-dom";
+import dayjs from "dayjs";
 import "../App.css";
 import "../grid.css";
 
 export default function Mentors(props) {
-	const [newQuizzQuestions, setNewQuizzQuestions]=useState([]);
-	const [refreshQuestions, setRefreshQuestions]=useState(false);
-
-	const [newQuizz,setNewQuizz]=useState(	{
-
+	const [newQuizQuestions, setNewQuizQuestions] = useState([]);
+	const [newQuiz, setNewQuiz] = useState({
 		name: "",
-		publiShingDate: "",
+		publishingDate: "",
 		questions_id: [],
 	});
 
-
-
 	useEffect(() => {
-
-
-
-		const makeQuestions=()=>{
-			let newQuizQuestions = [];
-			newQuizQuestions = newQuizz.questions_id.map((selId)=>{
-
-				let found = (props.questions.find((question)=>question._id==selId));
-				newQuizQuestions.push(found);
-				setNewQuizzQuestions(newQuizQuestions);
-			}
-			);
-
+		const makeQuestions = () => {
+			let selectedQuestions = [];
+			selectedQuestions = newQuiz.questions_id.map((selectedId) => {
+				let found = props.questions.find((question) => question._id === selectedId);
+				selectedQuestions.push(found);
+				setNewQuizQuestions(selectedQuestions);
+			});
 		};
 		makeQuestions();
-	},[newQuizz.questions_id]);
+	}, [newQuiz.questions_id, props.questions]);
```

Comment:
you don't actually need props.questions here, if terminal is coplaning it's not true, i bet you put this because he said about missing dependancy

Reply:
but props questions never change no point to put them here

Reply:
Yes, it's just to fix the warning